src/x/clock/offset_clock.go

```diff
@@ -35,7 +35,12 @@ func NewOffsetClock(offsetTime time.Time, nowFn NowFn) OffsetClock {
 	return OffsetClock{nowFn: nowFn, timeDelta: offsetTime.Sub(nowFn())}
 }
 
-// Now returns current time from the initial seed time value.
+// NewOffsetClockFromTimeDelta returns new offset clock that has time shifted by timeDelta duration.
+func NewOffsetClockFromTimeDelta(timeDelta time.Duration, nowFn NowFn) OffsetClock {
```

Comment:
Just wanted to say, IMHO, if this is the only constructor that we intend to use, then we don't really need to keep the other one around (with all the tests specific to it).